socs/agents/ibootbar/agent.py

```diff
@@ -280,7 +295,7 @@ def acq(self, session, params=None):
                 session.data = oid_cache
                 self.log.debug("{data}", data=session.data)
 
-                if get_result is None:
+                if not self.connected:
```

Comment:
I don't think we'd ever hit this now, since `self.connected` is set to `True` right before this `try:` block on line 286 now. We should just remove this `raise` at this point.

The `except` could be left in in a more general `except Exception as e` and log that exception, to avoid that `IndexError` causing a crash again, though I do expect the rest of this PR will avoid that anyway.